### PR TITLE
Photo comment authorization

### PR DESCRIPTION
**Pull request by YoungSong99**
Updated features:
- If user try to delete or edit a photo/comment that’s not user's, redirect them back and hide the link.
- user can visit any user’s profile and see followers/following, but posts and likes are visible only if:
The profile is not private.
- The profile is private, and user is an accepted follower.
- Only users can see their own pending follow requests.
- Feed and discover pages (/user/feed, /user/discover) are only viewable by that user.

### CODE REVIEW

File: app/controllers/comments_controller.rb
Comment (brvarner) [inline app/controllers/comments_controller.rb:1]: I love the way you formatted this else if statement for the destroy and edit actions!

File: app/controllers/users_controller.rb
Comment (brvarner) [inline app/controllers/users_controller.rb:1]: I like your solution to the problem a lot! It's a much better example of DRY than mine!

File: app/views/users/_user.html.erb
Comment (brvarner) [inline app/views/users/_user.html.erb:1]: Way to accomplish all the requirements! I would've missed this!

**Review by brvarner — APPROVED** [review]
Great work!

File: app/controllers/photos_controller.rb [hunk 1]
```diff
@@ -50,21 +52,35 @@ def update
 
   # DELETE /photos/1 or /photos/1.json
   def destroy
-    @photo.destroy
-    respond_to do |format|
-      format.html { redirect_back fallback_location: root_url, notice: "Photo was successfully destroyed." }
-      format.json { head :no_content }
+
+    if current_user == @photo.owner
```
Comment (rezahusain) [inline app/controllers/photos_controller.rb:56]: Good job making sure that the current user is the photo owner before allowing the user to delete!

**Review by rezahusain — APPROVED** [review]
Looks really good Young! Continue writing clean, concise code and you'll continue being successful.

**Review by CarlosCastanedaDev — APPROVED** [review]
Nice job, Young!

**Pull request merged**